lib/models/request.js

```diff
@@ -0,0 +1,46 @@
+var mongoose = require('mongoose'),
+Schema = require('mongoose').Schema,
+validators = require('mongoose-validators');
+
+var capitalize = function(val){
+  return val.toUpperCase();
+},
+lowercase = function(val){
+  return val.toLowerCase();
+};
+var Header = new Schema({
+  key : {
+    type : String
+  },
+  value : {
+    type : String
+  }
+}),
+Request = new Schema({
+  url: {
+    type: String,
+    required: true,
+    validate : validators.isURL()
+  },
+  method: {
+    type: String,
+    required: true,
+    "enum" : ['GET', 'POST', 'PUT', 'PATCH', 'DELETE', 'HEAD', 'OPTIONS'],
+    set: capitalize
+  },
+  // TODO: Should this be just part of the headers? 
+  contentType : {
```

Comment:
Yeah, saw that from the RFC - just didn't want to deal with case sensitivity in our array of headers, kinda out of laziness! 
I'm also of two minds as to if headers should be a separate schema, or freeform {"key" : "value" object...